### ISSUE TITLE
scheme.bed files coordinates are incorrect. 

### ISSUE THREAD

**Issue by ChrisgKent** [treangenlab/Olivar#14]
Hi,

When using the tool I've noticed that the coordinates for the bedfiles are off by one, when running;
```
olivar tiling --min-amp-len 1100 --max-amp-len 1300 -t 6 example_measles/olivar-ref.olvr

SA temperature = 0.000, SADDLE Loss = 22.939
Primer dimer optimization finished in 116.399s
Formatting design output...
Design file saved as ./6.olvd
Configurations saved as ./6.json
Primer pools saved as ./6.csv
Primer pools (ARTIC format) saved as ./6.scheme.bed
Reference sequence saved as ./6.fasta
Risk scores saved as ./6_risk.csv
Optimization plots saved as ./6_Loss.html
Risk and primer viewer saved as ./6.html
```

For the bedline
```
input-seq-1	45	76	6_1_LEFT	1	+	CTAGTACACTTAGGATTCAAGATCCTATTATC
input-seq-1	1276	1299	6_1_RIGHT	1	-	CAATCTCTGAAACAAGCCTTGCAT
```

```python
from Bio import SeqIO

ref = SeqIO.read("./git-issue/6.fasta", "fasta")

# For _LEFT primers
ref.seq[44:76]
> Seq('TAGTACACTTAGGATTCAAGATCCTATTATC')

# Need to move the left pos -1 to get correct slice
ref.seq[44-1:76]
> Seq('CTAGTACACTTAGGATTCAAGATCCTATTATC')

# For _RIGHT primers
ref.seq[1276:1299].reverse_complement()
> Seq('CAATCTCTGAAACAAGCCTTGCA')

# Need to move the left pos -1 to get correct slice
ref.seq[1276-1:1299].reverse_complement()
Seq('CAATCTCTGAAACAAGCCTTGCAT')

```

Right primers are a larger issue, as if primers are trimmed by these coords then a primer base will be left in the read. Leading to errors in concensus calling. 
[git-issue.zip](https://github.com/user-attachments/files/16714251/git-issue.zip)

**Comment by mxwang66** [comment]
Hi @ChrisgKent , 

Sorry for the late reply! So all the coordinates in Olivar are 1-based, closed interval, including those in the .bed files, for consistency. I'm not sure people are more used to 1-based or 0-based, closed interval or half open (python style), happy to take suggestions! 

Michael

**Comment by ChrisgKent** [comment]
Hi Michael,

Thanks for the response! 

I can’t speak for other use cases, however for .bed files, the expectation would be to follow the bed file format, which uses 0-based half open.

https://samtools.github.io/hts-specs/BEDv1.pdf

While reading the notes I did notice you stated all coord are 1 based. However, if people assume (as I did), they are typical beds, it will cause issues in downstream analysis.

Thanks! 
Chris
